datasetIngestor/syncDataToFileserver_unix.go

```diff
@@ -21,27 +23,49 @@ func SyncDataToFileserver(datasetId string, user map[string]string, RSYNCServer
 	// append trailing slash to sourceFolder to indicate that the *contents* of the folder should be copied
 	// no special handling for blanks in sourceFolder needed here
 	fullSourceFolderPath := sourceFolder + "/"
-	// check if filelisting given
-	// rsync can create only one level deep directory structure, here we need more, therefore mkdir -p
-	// This code is no longer needed, sine Edgar has a new rrsync wrapper which craetes the needed directory
-	// cmd := exec.Command("/usr/bin/ssh",RSYNCServer,"mkdir","-p",destFolder)
-	// // show rsync's output
-	// cmd.Stdout = os.Stdout
-	// cmd.Stderr = os.Stderr
-	//
-	// fmt.Printf("Running %v.\n", cmd.Args)
-	// cmd.Run()
-
-	cmd := exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxz", "--progress", "--msgs2stderr", fullSourceFolderPath, serverConnectString)
-	// // TODO: create folderstructure mkdir -p also for this case:
-	if absFileListing != "" {
-		cmd = exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxzr", "--progress", "--msgs2stderr", "--files-from", absFileListing, fullSourceFolderPath, serverConnectString)
+	
+	versionNumber, err := getRsyncVersion()
+	if err != nil {
+		log.Fatal("Error getting rsync version: ", err)
 	}
-	// show rsync's output
-	// cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	log.Printf("Running %v.\n", cmd.Args)
-	err = cmd.Run()
+	
+	// Check rsync version and adjust command accordingly
+	var rsyncCmd *exec.Cmd
```

Comment:
this feels a little as code duplicaction. To me what changes is only that in some cases we have to do `ssh -q`. What if we maybe just make this a variable and simplify the IF?

Reply:
(same for the --files-from, though slightly more complex, so we can keep it as is

Reply:
> this feels a little as code duplicaction. To me what changes is only that in some cases we have to do ssh -q. What if we maybe just make this a variable and simplify the IF?

it both adding/removing `-q` and replacing `--stderr=error` and `--msgs2stderr`

Reply:
but at least for the version check, one could simply do: 

if version.Compare(versionNumber, "3.2.3", ">=") { ssh="ssh -q"} else {ssh="ssh}

and then just pass this variable outside of the if to the exec command

Reply:
but at least for the version check, one could simply do:

if version.Compare(versionNumber, "3.2.3", ">=") { ssh="ssh -q"} else {ssh="ssh}

and then just pass this variable outside of the if to the exec command

Reply:
I've change them now